reflexible/data_structures.py

```diff
@@ -565,3 +565,284 @@ def max(self):
+
+        if self.ldirect == -1:
+            #backward run
+            tstart = self.sim_start - self.timedelta
+            tend = self.sim_end
+        elif self.ldirect == 1:
+            tstart = self.sim_start 
+            tend = self.sim_end + self.timedelta
+
+        with open(cfile, 'w') as outf:
+
+            outf.write('&COMMAND\n')
+            outf.write(' LDIRECT={0},\n'.format(self.ldirect))
+            outf.write(' IBDATE=  {0},\n'.format(tstart.strftime('%Y%m%d')))
+            outf.write(' IBTIME=  {0},\n'.format(tstart.strftime('%H%M%S')))
+            outf.write(' IEDATE=  {0},\n'.format(tend.strftime('%Y%m%d')))
+            outf.write(' IETIME=  {0},\n'.format(tend.strftime('%H%M%S')))
+            outf.write(' LOUTSTEP=             {0},\n'.format(self.loutstep))
+            outf.write(' LOUTAVER=             {0},\n'.format(self.loutaver))
+            outf.write(' LOUTSAMPLE=           {0},\n'.format(self.loutsample))
+            outf.write(' ITSPLIT=              {0},\n'.format(self.itsplit))
+            outf.write(' LSYNCTIME=            {0},\n'.format(self.lsynctime))
+            outf.write(' CTL=                  {0},\n'.format(self.ctl))
+            outf.write(' IFINE=                {0},\n'.format(self.ifine))
+            outf.write(' IOUT=                 {0},\n'.format(self.iout))
+            outf.write(' IPOUT=                {0},\n'.format(self.ipout))
+            outf.write(' LSUBGRID=             {0},\n'.format(self.lsubgrid))
+            outf.write(' LCONVECTION=          {0},\n'.format(self.lconvection))
+            outf.write(' LAGESPECTRA=          {0},\n'.format(self.lagespectra))
+            outf.write(' IPIN=                 {0},\n'.format(self.ipin))
+            outf.write(' IOUTPUTFOREACHRELEASE={0},\n'.format(self.ioutputforeachrelease))
+            outf.write(' IFLUX=                {0},\n'.format(self.iflux))
+            outf.write(' MDOMAINFILL=          {0},\n'.format(self.mdomainfill))
+            outf.write(' IND_SOURCE=           {0},\n'.format(self.ind_source))
+            outf.write(' IND_RECEPTOR=         {0},\n'.format(self.ind_receptor))
+            outf.write(' MQUASILAG=            {0},\n'.format(self.mquasilag))
+            outf.write(' NESTED_OUTPUT=        {0},\n'.format(self.nested_output))
+            outf.write(' LINIT_COND=           {0},\n'.format(self.linit_cond))
+            outf.write(' SURF_ONLY=            {0},\n'.format(self.surf_only))
+            outf.write(' CBLFLAG=              {0},\n/\n'.format(self.cblflag))
+            outf.close()
+
+
+
+class Ageclass(object):
+    """ General COMMAND input for Flexpart 
+
+    """
+    def __init__(self, ageclasses = [86400 * 50]):
+        self._keys = ['ageclasses']
+        for key in self._keys:
+            setattr(self, "_" + key, ageclasses)
+
+    def keys(self):
+        return self._keys
+
+    def __dir__(self):
+        """ necessary for Ipython tab-completion """
+        return self._keys
```

Comment:
Yes, IPython does not do auto-completion on variables starting on '_' because they are meant to be private (Python convention).  You must have a good reason to use variables that start with '_'.  Which is that?